clientconn.go

```diff
@@ -657,25 +658,36 @@ func (cc *ClientConn) handleResolvedAddrs(addrs []resolver.Address, err error) {
 		return
 	}
 
-	// TODO(bar switching) when grpclb is submitted, check address type and start grpclb.
-	if cc.balancerWrapper == nil {
-		// First time handling resolved addresses. Build a balancer use either
-		// the builder specified by dial option, or pickfirst.
-		builder := cc.dopts.balancerBuilder
-		if builder == nil {
-			// No customBalancer was specified by DialOption, and this is the first
-			// time handling resolved addresses, create a pickfirst balancer.
-			builder = newPickfirstBuilder()
+	cc.curAddresses = addrs
+
+	if cc.dopts.balancerBuilder != nil && cc.balancerWrapper == nil {
+		cc.balancerWrapper = newCCBalancerWrapper(cc, cc.dopts.balancerBuilder, cc.balancerBuildOpts)
+	} else {
+		newBalancerName := cc.curBalancerName
+		// This is the first time handling resolved addresses, try to create a
+		// pickfirst balancer.
+		if newBalancerName == "" {
+			newBalancerName = pickfirstName
+		}
+		for _, a := range addrs {
+			if a.Type == resolver.GRPCLB && newBalancerName != grpclbName {
```

Comment:
I think we don't want "`newBalancerName != grpclbName`".  We should `break` if any `Type` is `resolver.GRPCLB`, and re-assigning newBalancerName to grpclbName is not worth optimizing away (actually it's more optimal than the extra branch).